File: 3513-number-of-unique-xor-triplets-i/solution.cpp
#include <vector>

class Solution {
public:
    int uniqueXorTriplets(std::vector<int>& nums) {
        int n = nums.size();
        
        // Base cases
        if (n == 1) return 1;
        if (n == 2) return 2;
        
        // For n >= 3, the answer is 2^(bit_length(n))
        int bit_len = 32 - __builtin_clz(n);
        return 1 << bit_len;
    }
};

File: 3513-number-of-unique-xor-triplets-i/solution.py
class Solution:
    def uniqueXorTriplets(self, nums: list[int]) -> int:
        n = len(nums)
        
        # Base cases
        if n == 1:
            return 1
        if n == 2:
            return 2
        
        # For n >= 3, the answer is 2^(bit_length(n))
        return 1 << n.bit_length()